### Ecological Genomics Proposal

#### Team Sherlock

**February 1, 2017**

<u>**The Opener**</u>: "The 2013 sea star wasting disease outbreak is the largest documented marine epizootic disease of non-commercial marine animals." (Eisenlord et al. 2016) Over 20 species of sea stars were devastated by the disease; can reach up to 100% prevalence;  

 Mess up marine community

<u>**Current Knowledge**</u>: 

* Purple sea urchin immune response genes (Hibino et al. 2006)
* Immune response of *Pycnopodia helianthoides* to sea star wasting disease (Fuess et al. 2015)
* Antibacterial response of a sea urchin (*Heliocidaris erythrogramma*) (Dheilly et al. 2011)
* Densovirus associated with sea-star wasting disease and mass mortality (Hewson et al. 2014)

<u>**The Gap Statement**</u>: 

* The densovirus does not appear to be associated with these symptomatic individuals, so the causative pathogen is unknown. In order to prevent further sea star devastation, the causative agent must be identified.
* Determing whether certain immune response genes are being expressed is critical to understanding what is causing these mortalities. 



<u>**Objectives**</u>:  

<u>**Central Hypothesis**</u>: 

How will we test this hypothesis? How will we determine between alternative explanations? Will we have the statistical power to do so?

* Lit review to identify potential immune response genes
* 5 individuals for each group: HH, HS, SS; 2 time steps for H->S transition and corresponding HH and SS days
* Look for difference in expression (# reads)
* Groups of random genes to compare background variation to our targeted variation 

What are your expected results/outcome (adress objectives/fill the gap)?

