#Ecological Genomics Spring 2017    

## Author: Lauren Ash      

##Overall Description of notebook      

This is my online notebook for the Ecological Genomics course. I will update with new techniques I learn, questions that I come up with, and general notes. 


##Date started: 2017-01-19 
##Date end:  -   

##Philosophy   
Science should be reproducible and one of the best ways to achieve this is by logging research activities in a notebook. Because science/biology has increasingly become computational, it is easier to document computational projects in an electronic form, which can be shared online through Github.

###Helpful features of the notebook     


**It is absolutely critical for your future self and others to follow your work.**     

* The notebook is set up with a series of internal links from the table of contents.    
* All notebooks should have a table of contents which has the "Page", date, and title (information that allows the reader to understand your work).     
* Also, one of the perks of keeping all activities in a single document is that you can **search and find elements quickly**.     
* You can document anything you'd like, aside from logging your research activities. For example:
	* feel free to log all/any ideas for your research project([example](https://github.com/adnguyen/Notebooks_and_Protocols/blob/master/2016_notebook.md#page-39-2016-06-13-post-doc-project-idea-assessing-current-impacts-of-climate-change-in-natural-populations)) as an entry,     
	* or write down notes for a paper([example](https://github.com/adnguyen/Notebooks_and_Protocols/blob/master/2016_notebook.md#id-section36).      

* Lastly, you can share specific entries because of the three "#" automatically creates a link when the notebook renders on github. 

<a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Creative Commons License" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" /></a><br />This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.


###Table of contents for 60 entries (Format is *Page: Date(with year-month-day). Title*)        
* [Page 1: 2017-01-19](#id-section1). First journal article: Ellegren 2014 
* [Page 2: 2017-01-23](#id-section2). Class notes [MISSED]
* [Page 3: 2017-01-24](#id-section3). Articles: Rockman 2012 and Lee et al. 2014
* [Page 4: 2017-01-25](#id-section4). Class notes
* [Page 5:](#id-section5).
* [Page 6:](#id-section6).
* [Page 7:](#id-section7).
* [Page 8:](#id-section8).
* [Page 9:](#id-section9).
* [Page 10:](#id-section10).
* [Page 11:](#id-section11).
* [Page 12:](#id-section12).
* [Page 13:](#id-section13).
* [Page 14:](#id-section14).
* [Page 15:](#id-section15).
* [Page 16:](#id-section16).
* [Page 17:](#id-section17).
* [Page 18:](#id-section18).
* [Page 19:](#id-section19).
* [Page 20:](#id-section20).
* [Page 21:](#id-section21).
* [Page 22:](#id-section22).
* [Page 23:](#id-section23).
* [Page 24:](#id-section24).
* [Page 25:](#id-section25).
* [Page 26:](#id-section26).
* [Page 27:](#id-section27).
* [Page 28:](#id-section28).
* [Page 29:](#id-section29).
* [Page 30:](#id-section30).
* [Page 31:](#id-section31).
* [Page 32:](#id-section32).
* [Page 33:](#id-section33).
* [Page 34:](#id-section34).
* [Page 35:](#id-section35).
* [Page 36:](#id-section36).
* [Page 37:](#id-section37).
* [Page 38:](#id-section38).
* [Page 39:](#id-section39).
* [Page 40:](#id-section40).
* [Page 41:](#id-section41).
* [Page 42:](#id-section42).
* [Page 43:](#id-section43).
* [Page 44:](#id-section44).
* [Page 45:](#id-section45).
* [Page 46:](#id-section46).
* [Page 47:](#id-section47).
* [Page 48:](#id-section48).
* [Page 49:](#id-section49).
* [Page 50:](#id-section50).
* [Page 51:](#id-section51).
* [Page 52:](#id-section52).
* [Page 53:](#id-section53).
* [Page 54:](#id-section54).
* [Page 55:](#id-section55).
* [Page 56:](#id-section56).
* [Page 57:](#id-section57).
* [Page 58:](#id-section58).
* [Page 59:](#id-section59).
* [Page 60:](#id-section60).  

------ <div id='id-section1'/>
###Page 1: 2017-01-19. First journal article: Ellegren 2014

Here are [Andrew's thoughts](https://github.com/adnguyen/2017_Ecological_Genomics/blob/master/ANBE_notebook.md#page-2-2017-01-20-readings-for-2017-01-25-monday) on the paper "Genome sequencing and population genomics in non-model organisms" (under the heading **Now paper notes:**).   

### My thoughts while reading the paper

The paper has a glossary of terms and I wanted to write a few down:
   
**Effective population size (N<sub>e</sub>):** a measure of the size of an idealized population in which the effect of genetic drift on allele frequencies is similar to the population under consideration.   
   
**Genetic architecture:** the genetic background to phenotypic traits, including their number, effect sizes, and dominance.   
   
**Genome-wide association studies (GWAS):** studies based on the use of large numbers of SNP markers genotyped in a group showing a particular trait, and in a control group, with the aim of finding association between trait and markers.   

**Scaled selection coefficient:** the selection coefficient (*s*, the relative fitness dis-/advantage of a derived allele) multiplied with N<sub>e</sub>, to take into account the fact that the efficiency of selection is directly proportional to population size. 
   
**Transcriptome:** the complete repertoire of transcribed sequences in the genome, including expression of both from protein-coding genes and from noncoding RNAs.   
 
Some parts that stood out to me:  

* **Adaptive evolution** should be more common in those regions of the genome experiencing high rates of recombination.   
* **GC-biased gene conversion** is a process in which C and G nucleotides have a higher probability of being the donor -> genomic regions with high recombination rates should evolve towards a high GC content -> explains heterogeneous landscape of base composition   
* Not fully explored: biased gene conversion has potential to reduce efficacy of purifying selection in high recombination regions by aggravating the removal of deleterious AT/CG mutations.   
* Being able to quantify the proportion of the genome evolving under purifying selection is important (identifying conserved sequences) * There is a debate about what contributes to the 'functionality' of the genome (transcribed sequence, histone modification, open chromain indicative of function vs only proven purifying selection)
* When **scaled selection coefficents** are higher, one might expect to see larger fractions of the genome conserved   
* Having genome access, we can conclude gene duplication and neofunctionalization represent common mechanisms for formation of new phenotypes and thereby a seedbed for adaptation.   
* Adaptive evolution can be estimated by comparing the rate of nucleotide substitutions (likely to have functional consequences) vs presumably neutral substitutions, but it is hard to estimate in small populations.   
* **Population genomics:** the use of sequence or genotype data from multiple, although individually analyzed, loci spread across the genome; scales up inference power AND offers a means to study the genomic landscape and variance of allelic diversity within and between populations.   
	+ Typical pipeline:   
		1. Design of sequencing strategy (# of individuals per population; # of populations; etc)    
		2. Generation of sequence data   
		3. Mapping of sequence reads to the assembly   
		4. Variant calling (genotyping)   
		5. Downstream population genetic or molecular evolutionary analyses  
* **Trait mapping** example: resequenced pigeon genomes with and without a head crest phenotype and identified a particular allelic variant present in the crested breeds, absent in the others.   
* Regions with low diversity within populations are indicative of recent selective sweeps and regions with high divergence between populations point to diversifying selection and limited gene exchange

**Overall messages:**  
* Having access to genomes is letting us infer evolutionary processes affecting sequence evolution from the whole-genome perspective instead of a random sample of loci.   
* The technology is rapidly evolving (next generation sequencing is no longer a fitting name)
* Population genomics is becoming a venue for the identification of genes/genomic regions involved in fitness-related traits and speciation.   
   
**How can I apply to my research?**   
* Trait mapping? Could compare genomes of asymptomatic and symptomatic ranavirus-infected individuals and seeing if there are allelic variants


------ <div id='id-section2'/>
###Page 2: 2017-01-23 Course notes [MISSED]

Andrew posted notes of the class and Melissa's info update that I missed [here](https://github.com/adnguyen/2017_Ecological_Genomics/blob/master/ANBE_notebook.md#page-3-2017-01-23-day-2-course-notes).   

Info I looked up after reading Andrew's notes:   
   
Sanger sequencing: "The first method of sequencing the genetic code was devised by Fred Sanger. To sequence the DNA, it must first be separated into two strands. The strand to be sequenced is copied using chemically altered bases. These altered bases cause the copying process to stop each time one particular letter is incorporated into the growing DNA chain. This process is carried out for all four bases, and then the fragments are put together like a jigsaw to reveal the sequence of the original piece of DNA." [Reference](https://www.dnalc.org/view/15479-Sanger-method-of-DNA-sequencing-3D-animation-with-narration.html)   
   
Illumina: Illumina next-generation sequencing utilizes a fundamentally different approach from the classic Sanger chain-termination method. It leverages sequencing by synthesis (SBS) technology – tracking the addition of labeled nucleotides as the DNA chain is copied – in a massively parallel fashion. [Reference](https://www.illumina.com/technology/next-generation-sequencing.html)      

------ <div id='id-section3'/>
###Page 3: 2017-01-24 Articles: Rockman 2012 and Lee et al. 2014
   
### Lee et al. 2014: Identifying the genes underlying quantitative traits: a rationale for the QTN programme   
   
**QTN: quantitative trait nucleotide**      
* Goal = identifying the genes or nucleotide variants underlying quantitative and adaptive traits   
* Criticisms   
	+ QTN programme has asserted that finding the genes and nucleotides for adaptive/quant. traits is a goal without explaining why it is a goal   
	+ Logical flaws in how results are interpreted (i.e. overestimating phenotypic consequences; biased picture of importance)   
	+ Holds little value: focuses on patterns instead of evolutionary processes
* Exciting: adaptive phenotypes can be understood as a collection of single-nucleotide polymorphisms (SNPs)   

**Rationale for QTN**   
* Vertical integration   
	+ Understanding the evolution of adaptation from genetic variation at causal nucleotides to -> heritablity in natural populations -> strength of contemporary selection -> how patterns of genetic variation and selection depend on geography/ecological context
	+ Suprising in the lack of complete vertical integration examples   
	+ Example: Hopkins and Rauscher 2011 - found nucleotide diversity patterns indicative of recent, strong natural selection and investigated how pollinators affect intensity/color of flowers   
	+ Few examples exist where selection on QTN for truly quantitative traits has been demonstrated (focus on qualitative like pink vs red flowers)   
* Genetic parallelism and pleitropy   
	+ QTN research can reveal underlying mechanisms of convergent phenotypic evolution across related organismal groups   
	+ Only way to directly test for the extension of convergence down to the nucleotide level   
	+ Production of common phenotype can imply either i) there are developmental/structural constraints on the production of the trait or ii) repeated changes are favored because they have fewer deleterious pleiotropic side-effects relative to others
* Maintenance of standing genetic variation    
	+ Why is there such an abundance of heritable variation in nature?   
	+ Mutation-selection balance: equilibrium between mutation introducing deleterious variation and purifying selection depleting it; variants are mostly rare, recessive alleles   
	+ Balancing selection hypothesis: negative and positive selection acting on same site for different components of a trait; result in maintenance of both alleles and predicts polymorphisms at intermediate frequencies   
	+ QTN programme can examine how selection acts on standing variation, thereby connecting theory, quantitative genetics, and population genetics.
* Role of standing genetic variation in the wild    
	+ Evolutionary responses to selection will differ greatly depending on whether the 'substrate of adaptation' comes from new mutations or standing variance   
	+ Adaptation from new mutation is accompanied by the signature of a 'hard sweep' (steep decline in nucleotide diversity around a fixed site; excess of rare alleles close to selected site)   
	+ Adaptation from standing variance is accompanied by the signature of a 'soft sweep' (lesser decline in diversity and associated with intermediate-frequency alleles)   
* Understanding the role of genomic architecture in adaptation   
	+ QTN programme can clarify the role of aspects of genomic architecture (chromosomal inversions, translocations, 'supergenes' in areas of suppressed/restricted recombination) in adaptation   
	+ Can clarify genetic basis and evolutionary forces responsible for polymorphisms (i.e. that directly influence plant mating and fitness)   
	+ Necessary for evaluating role of genomic architecture in studies on role of chromosomal inversions in local adaptation, which could 'capture' locally advantageous haplotypes and spread quite rapidly  
	+ Ex: monkeyflower - could estimate the relative contribution of inversion to reproductive isolation between the two ecotypes   
	
**Take away:** We should refocus and redouble our efforts to gain insight into the evolutionary process, use effective genetic data, and test evolutionary hypotheses	
	
**How can this relate to my research?**   
* Could look at different virus strains? Try to relate nucleotides to adaptation in virulence?   

------ <div id='id-section4'/>
###Page 4: 2017-01-25 Class notes
------ <div id='id-section5'/>
###Page 5:
------ <div id='id-section6'/>
###Page 6:
------ <div id='id-section7'/>
###Page 7:
------ <div id='id-section8'/>
###Page 8:
------ <div id='id-section9'/>
###Page 9:
------ <div id='id-section10'/>
###Page 10:
------ <div id='id-section11'/>
###Page 11:
------ <div id='id-section12'/>
###Page 12:
------ <div id='id-section13'/>
###Page 13:
------ <div id='id-section14'/>
###Page 14:
------ <div id='id-section15'/>
###Page 15:
------ <div id='id-section16'/>
###Page 16:
------ <div id='id-section17'/>
###Page 17:
------ <div id='id-section18'/>
###Page 18:
------ <div id='id-section19'/>
###Page 19:
------ <div id='id-section20'/>
###Page 20:
------ <div id='id-section21'/>
###Page 21:
------ <div id='id-section22'/>
###Page 22:
------ <div id='id-section23'/>
###Page 23:
------ <div id='id-section24'/>
###Page 24:
------ <div id='id-section25'/>
###Page 25:
------ <div id='id-section26'/>
###Page 26:
------ <div id='id-section27'/>
###Page 27:
------ <div id='id-section28'/>
###Page 28:
------ <div id='id-section29'/>
###Page 29:
------ <div id='id-section30'/>
###Page 30:
------ <div id='id-section31'/>
###Page 31:
------ <div id='id-section32'/>
###Page 32:
------ <div id='id-section33'/>
###Page 33:
------ <div id='id-section34'/>
###Page 34:
------ <div id='id-section35'/>
###Page 35:
------ <div id='id-section36'/>
###Page 36:
------ <div id='id-section37'/>
###Page 37:
------ <div id='id-section38'/>
###Page 38:
------ <div id='id-section39'/>
###Page 39:
------ <div id='id-section40'/>
###Page 40:
------ <div id='id-section41'/>
###Page 41:
------ <div id='id-section42'/>
###Page 42:
------ <div id='id-section43'/>
###Page 43:
------ <div id='id-section44'/>
###Page 44:
------ <div id='id-section45'/>
###Page 45:
------ <div id='id-section46'/>
###Page 46:
------ <div id='id-section47'/>
###Page 47:
------ <div id='id-section48'/>
###Page 48:
------ <div id='id-section49'/>
###Page 49:
------ <div id='id-section50'/>
###Page 50:
------ <div id='id-section51'/>
###Page 51:
------ <div id='id-section52'/>
###Page 52:
------ <div id='id-section53'/>
###Page 53:
------ <div id='id-section54'/>
###Page 54:
------ <div id='id-section55'/>
###Page 55:
------ <div id='id-section56'/>
###Page 56:
------ <div id='id-section57'/>
###Page 57:
------ <div id='id-section58'/>
###Page 58:
------ <div id='id-section59'/>
###Page 59:
------ <div id='id-section60'/>
###Page 60:

------
